R/setops.R

```diff
@@ -103,8 +103,8 @@ fsetequal = function(x, y, all=TRUE) {
 # all.equal ----
 
 all.equal.data.table = function(target, current, trim.levels=TRUE, check.attributes=TRUE, ignore.col.order=FALSE, ignore.row.order=FALSE, tolerance=sqrt(.Machine$double.eps), ...) {
-  stopifnot(is.logical(trim.levels), is.logical(check.attributes), is.logical(ignore.col.order), is.logical(ignore.row.order), is.numeric(tolerance))
-  if (!is.data.table(target) || !is.data.table(current)) stop("'target' and 'current' must both be data.tables")
+  stopifnot(is.logical(trim.levels), is.logical(check.attributes), is.logical(ignore.col.order), is.logical(ignore.row.order), is.numeric(tolerance), is.data.table(target))
+  if (!is.data.table(current)) return(NextMethod())
```

Comment:
why dispatch to next method instead of returning scalar character about class mismatch? if `target` is a data.table then it make sense to produce an answer in data.table method rather than redirecting to another.

Reply:
I guess I thought that `all.equal(DT, DF)` should return roughly the same info as `all.equal(DF, DT)`. 

But happy to do the just do that if you prefer.

Reply:
I would rather prefer to return scalar early rather than investigating all differences, `all.equal` can be expensive, especially when you use `ignore.row.order`. Of course `check.attributes=FALSE` should continue to work, the same way as `all.equal(DF, DT, check.attributes=FALSE)` do.